File: src/aiperf/cli_commands/validate.py
# SPDX-FileCopyrightText: Copyright (c) 2025-2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
# SPDX-License-Identifier: Apache-2.0
"""CLI command for validating benchmark artifacts."""

from __future__ import annotations

from pathlib import Path
from typing import Annotated, Literal

from cyclopts import App, Parameter

app = App(name="validate")


@app.default
def validate(
    target: Annotated[
        Literal["mooncake-trace"],
        Parameter(help="Artifact format to validate"),
    ],
    input: Path,
) -> None:
    """Validate a benchmark artifact.

    Args:
        target: Artifact format to validate.
        input: Path to the artifact file.
    """
    match target:
        case "mooncake-trace":
            from aiperf.dataset.agentic_code_gen.cli import validate as _validate

            _validate(input=input)

File: rust/cli/tests/validate.rs
// SPDX-FileCopyrightText: Copyright (c) 2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
// SPDX-License-Identifier: Apache-2.0
//! `aiperf validate mooncake-trace` command contract coverage.

use std::process::Command;

fn fixture(name: &str) -> String {
    format!(
        "{}/../../tools/parity/validate/{name}",
        env!("CARGO_MANIFEST_DIR")
    )
}

#[test]
fn validate_pass_is_byte_exact() {
    let out = Command::new(env!("CARGO_BIN_EXE_aiperf"))
        .args([
            "validate",
            "mooncake-trace",
            "--input",
            &fixture("valid.jsonl"),
        ])
        .output()
        .expect("run");
    assert!(out.status.success());
    assert_eq!(
        String::from_utf8_lossy(&out.stdout),
        "Validation passed: 5 rows are Mooncake-compatible.\n"
    );
}

#[test]
fn validate_fail_flags_all_rows() {
    let out = Command::new(env!("CARGO_BIN_EXE_aiperf"))
        .args([
            "validate",
            "mooncake-trace",
            "--input",
            &fixture("invalid.jsonl"),
        ])
        .output()
        .expect("run");
    assert_eq!(out.status.code(), Some(1));
    let s = String::from_utf8_lossy(&out.stdout);
    assert!(
        s.starts_with("Validation failed with 4 error(s):"),
        "got: {s}"
    );
}
